src/pineko/fonll.py

```diff
@@ -182,6 +182,8 @@ def produce_combined_fk(
     "FONLL-A": [1 for _ in range(7)],
     "FONLL-B": [2, 2, 1, 2, 2, 1, 2],
     "FONLL-C": [2 for _ in range(7)],
+    "FONLL-D": [3, 3, 2, 3, 3, 2, 3],
+    "FONLL-E": [3 for _ in range(7)],
```

Comment:
All the lists above could be generated from a single tuple:

```py
def components_orders(fo, res):
    return [fo, fo, resum, fo, fo, resum, fo]
```

But I'm not sure about the meaning of `"full"` (because I suspect you do not even need the 7 elements).

Reply:
Why not?

Reply:
Because there is a pattern, and I guess I know where this pattern is coming from.

Reply:
No wait my question was, why do you think we do not even need the 7 elements?

Reply:
Because there is a pattern, and in that case you might be able to make a function respecting that pattern, without making a list.

Reply:
Okok I agree about this of course. I believe I misunderstood your first comment.